frontend/src/types.ts

```diff
@@ -22,7 +22,7 @@ export type PrometheusQueryRangeResponse = {
       {
         // not used -- see https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries for more info
         metric: unknown;
-        values: PrometheusQueryRangeResultValue[];
+        values: PrometheusQueryRangeResultValue[] | undefined;
```

Comment:
As far as I can tell, `values` is always an array.

The bigger issue is that `PrometheusQueryRangeResponse` as a whole may not include `data` or or it may be some other object. It depends on the `status` of the call. If `status` is `success` the `data` is as expected here. However if `status` is `failure` data may be something else (although the docs aren't clear). We should probably create a union type for the different `status` values.